webapp/src/Controller/Jury/ShadowDifferencesController.php

```diff
@@ -0,0 +1,207 @@
+<?php declare(strict_types=1);
+
+namespace App\Controller\Jury;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query\Expr\Join;
+use App\Controller\BaseController;
+use App\Entity\ExternalJudgement;
+use App\Entity\Judging;
+use App\Entity\Submission;
+use App\Service\DOMJudgeService;
+use App\Service\SubmissionService;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/jury/shadow-differences")
+ * @IsGranted("ROLE_ADMIN")
+ */
+class ShadowDifferencesController extends BaseController
+{
+    /**
+     * @var DOMJudgeService
+     */
+    protected $dj;
+
+    /**
+     * @var SubmissionService
+     */
+    protected $submissions;
+
+    /**
+     * @var SessionInterface
+     */
+    protected $session;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    protected $em;
+
+    public function __construct(
+        DOMJudgeService $dj,
+        SubmissionService $submissions,
+        SessionInterface $session,
+        EntityManagerInterface $em
+    ) {
+        $this->dj          = $dj;
+        $this->submissions = $submissions;
+        $this->session     = $session;
+        $this->em          = $em;
+    }
+
+    /**
+     * @Route("", name="jury_shadow_differences")
+     */
+    public function indexAction(Request $request)
+    {
+        $shadowMode = DOMJudgeService::DATA_SOURCE_CONFIGURATION_AND_LIVE_EXTERNAL;
+        $dataSource = $this->dj->dbconfig_get('data_source', DOMJudgeService::DATA_SOURCE_LOCAL);
+        if ($dataSource != $shadowMode) {
+            $this->addFlash('danger', sprintf(
+                'Shadow differences only supported when data_source is %d',
+                $shadowMode
+            ));
+            return $this->redirectToRoute('jury_index');
+        }
+
+        // Close the session, as this might take a while and we don't need the session below
+        $this->session->save();
+
+        $contest        = $this->dj->getCurrentContest();
+        $verdictsConfig = $this->dj->getDomjudgeEtcDir() . '/verdicts.php';
+        $verdicts       = array_merge(['judging' => 'JU'], include $verdictsConfig);
```

Comment:
Kattis also has JE/judge error - how would we process/display that?

Reply:
Currently we don't, because the import command only accepts stuff from `verdicts.php`. We should fix this at a later point I guess, but not sure how yet.

Reply:
Perhaps leave a TODO.

Reply:
I have added it to the import command